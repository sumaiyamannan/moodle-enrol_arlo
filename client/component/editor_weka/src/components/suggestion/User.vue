<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Kian Nguyen <kian.nguyen@totaralearning.com>
  @module editor_weka
-->

<template>
  <div class="tui-wekaUserSuggestions" :style="positionStyle">
    <Dropdown
      :separator="true"
      :open="$apollo.loading || users.length > 0"
      @dismiss="$emit('dismiss')"
    >
      <span class="sr-only">
        {{ $str('matching_users', 'editor_weka') }}:
      </span>

      <template v-if="$apollo.loading">
        <DropdownItem :disabled="true">
          {{ $str('loadinghelp', 'moodle') }}
        </DropdownItem>
      </template>

      <DropdownItem
        v-for="(user, index) in users"
        :key="index"
        :no-padding="true"
        @click="pickUser(user)"
      >
        <MiniProfileCard :no-border="true" :display="user.card_display" />
      </DropdownItem>
    </Dropdown>
  </div>
</template>

<script>
import Dropdown from 'tui/components/dropdown/Dropdown';
import DropdownItem from 'tui/components/dropdown/DropdownItem';
import findUsers from 'editor_weka/graphql/find_users_by_pattern';
import MiniProfileCard from 'tui/components/profile/MiniProfileCard';

export default {
  components: {
    Dropdown,
    DropdownItem,
    MiniProfileCard,
  },

  props: {
    contextId: {
      type: [Number, String],
    },

    component: {
      type: String,
    },

    area: {
      type: String,
    },

    x: {
      // This is offset from left
      required: true,
      type: [Number, String],
    },

    y: {
      // This is offset from top.
      required: true,
      type: [Number, String],
    },

    pattern: {
      required: true,
      type: String,
    },
  },

  apollo: {
    users: {
      query: findUsers,
      fetchPolicy: 'network-only',
      variables() {
        return {
          pattern: this.pattern,
          contextid: this.contextId,
          component: this.component,
          area: this.area,
        };
      },
    },
  },

  data() {
    return {
      users: [],
    };
  },

  computed: {
    positionStyle() {
      return {
        left: `${this.x}px`,
        top: `${this.y}px`,
      };
    },
  },

  methods: {
    /**
     *
     * @param {Number} id
     * @param {String} fullname
     */
    pickUser({ id, fullname }) {
      this.$emit('item-selected', { id, text: fullname });
    },
  },
};
</script>

<lang-strings>
  {
    "editor_weka": [
      "matching_users"
    ],
    "moodle": [
      "loadinghelp"
    ]
  }
</lang-strings>

<style lang="scss">
.tui-wekaUserSuggestions {
  position: absolute;
  width: 32.6rem;
}
</style>
