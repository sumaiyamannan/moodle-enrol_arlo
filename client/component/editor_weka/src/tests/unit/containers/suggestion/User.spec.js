/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Kian Nguyen <kian.nguyen@totaralearning.com>
 * @module editor_weka
 */

import User from 'editor_weka/components/suggestion/User';
import { shallowMount } from '@vue/test-utils';

describe('editor_weka/components/suggestion/User.vue', function() {
  let wrapper = null;

  beforeAll(function() {
    wrapper = shallowMount(User, {
      mocks: {
        $apollo: {
          loading: false,
        },
      },

      propsData: {
        location: { x: 1, y: 1 },
        pattern: 'bolobala',
      },
    });
  });

  it('Checks snapshot', function() {
    expect(wrapper.element).toMatchSnapshot();
  });
});
