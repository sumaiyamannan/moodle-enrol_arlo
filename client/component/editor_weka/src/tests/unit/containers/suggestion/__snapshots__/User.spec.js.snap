// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editor_weka/components/suggestion/User.vue Checks snapshot 1`] = `
<div
  class="tui-wekaUserSuggestions"
  style="left: 1px; top: 1px;"
>
  <dropdown-stub
    animation="default"
    canclose="true"
    closeonclick="true"
    inlinemenu="true"
    position="bottom-left"
    role="menu"
    separator="true"
  >
    <span
      class="sr-only"
    >
      
      [[matching_users, editor_weka]]:
    
    </span>
     
    <!---->
     
  </dropdown-stub>
</div>
`;
